### ISSUE TITLE
Buffer overflow in `snprintf` in vmcircbuf_mmap

### ISSUE THREAD

**Issue by marcusmueller** [gnuradio/gnuradio#2709]
```
/buildbot/gnuradio/gnuradio-runtime/lib/vmcircbuf_mmap_tmpfile.cc: In constructor ‘gr::vmcircbuf_mmap_tmpfile::vmcircbuf_mmap_tmpfile(int)’:
/buildbot/gnuradio/gnuradio-runtime/lib/vmcircbuf_mmap_tmpfile.cc:79:53: warning: ‘%s’ directive output may be truncated writing up to 1023 bytes into a region of size 990 [-Wformat-truncation=]
   79 |                  "gr::vmcircbuf_mmap_tmpfile: open [%s]", seg_name);
      |                                                     ^~    ~~~~~~~~
/buildbot/gnuradio/gnuradio-runtime/lib/vmcircbuf_mmap_tmpfile.cc:78:17: note: ‘snprintf’ output between 36 and 1059 bytes into a destination of size 1024
   78 |         snprintf(msg, sizeof (msg),
      |         ~~~~~~~~^~~~~~~~~~~~~~~~~~~
   79 |                  "gr::vmcircbuf_mmap_tmpfile: open [%s]", seg_name);
```

**Comment by marcusmueller** [comment]
closed by #2807

**Comment by marcusmueller** [comment]
closed by #3339
